File: articles.jdbc/src/main/java/pet/db/jdbc/repository/AuthorshipOfArticleRepository.java
package pet.db.jdbc.repository;

import jakarta.validation.constraints.NotNull;

import pet.db.jdbc.entity.AuthorshipOfArticle;

import java.util.List;

public interface AuthorshipOfArticleRepository {

    List<AuthorshipOfArticle> save(@NotNull List<AuthorshipOfArticle> authorshipOfArticles);

    List<Integer> findAuthorIdsByArticleId(@NotNull Integer articleId);

    List<Integer> findArticleIdsByAuthorId(@NotNull Integer authorId);

    boolean exists(@NotNull AuthorshipOfArticle authorshipOfArticle);

}

File: src/main/kotlin/pet/articles/repository/authorship/AuthorshipOfArticleRepository.kt
package pet.articles.repository.authorship

import pet.articles.model.dto.AuthorshipOfArticle

interface AuthorshipOfArticleRepository {

    fun findAuthorIdsByArticleId(articleId: Int): List<Int>

    fun findArticleIdsByAuthorId(authorId: Int): List<Int>

    fun exists(authorshipOfArticle: AuthorshipOfArticle): Boolean
}
